packages/api/src/auth/decoders/clerk.ts

```diff
@@ -1,38 +1,28 @@
-import type { APIGatewayProxyEvent, Context as LambdaContext } from 'aws-lambda'
+import type IClerk from '@clerk/clerk-sdk-node/instance'
 
-interface Req {
-  event: APIGatewayProxyEvent
-  context: LambdaContext
-}
-
-export const clerk = async (token: string, req: Req) => {
+export const clerk = async (token: string) => {
```

Comment:
Is this to still support older clerk applications? Or are there scenarios where new clerk apps also would have the token in the cookie?

Reply:
If we want to add cookie support, we should pull in `getAuthState` and the interstitial instead of this naive check.

Context for the Redwood crew...

Clerk has the capability to perform auth based on a cookie named `__session` or the `Authorization` header.  Both strategies use the exact same JWT value.

The advantage of the cookie is that it works with SSR or direct browser navigation to an API route.

But, it's more complicated because we need to solve for direct visits when the JWT has expired (this is never a problem for the `Authorization` header because we call `getToken` immediately before any request).

When an app is using cookie auth and the JWT has expired, we throw an intermediary page to refresh the cookie (the "interstitial"), then reload the request.  This logic for deciding when to throw the interstitial is baked into the `getAuthState` function I'm referencing above.

Unless you're planning SSR in the immediate future (where cookie auth is strictly necessary), we can skip this for now.

Reply:
@igneel64 @colinclerk So what's the consensus here? 😅 

Leave it as it is and add cookie support (using `getAuthState`) later?

Reply:
Sure, we can start with the header token here, let's also skip the cookie for now :)